From 7601c291d3d3267288f334a4cede41bd1d0b850a Mon Sep 17 00:00:00 2001
From: root <root@softwaresecurity.(none)>
Date: Thu, 11 Apr 2013 21:18:28 +0200
Subject: [PATCH 03/23] Ignore swap files

---
 .gitignore |    1 +
 1 file changed, 1 insertion(+)
 create mode 100644 .gitignore

diff --git a/.gitignore b/.gitignore
new file mode 100644
index 0000000..1377554
--- /dev/null
+++ b/.gitignore
@@ -0,0 +1 @@
+*.swp
-- 
1.7.9.5

