From d1f7a84cf2c784bf4596c0508dd1839b5a986350 Mon Sep 17 00:00:00 2001
From: root <root@softwaresecurity.(none)>
Date: Thu, 11 Apr 2013 23:55:07 +0200
Subject: [PATCH 15/23] Hide the password as the user is entering it

---
 profile.php |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/profile.php b/profile.php
index 8a09ea9..beca93c 100644
--- a/profile.php
+++ b/profile.php
@@ -60,7 +60,7 @@ $page[ 'body' ] .= "This is your user Profile<br />
     First Name *<br /><input name=\"txtFirst\" type=\"text\" size=\"30\" maxlength=\"15\" value=\"". $fn . "\"><br/>
     Last Name *<br /><input name=\"txtLast\" type=\"text\" size=\"30\" maxlength=\"15\" value=\"". $ln . "\"><br/>
     Username *<br /><input name=\"txtUser\" type=\"text\" size=\"30\" maxlength=\"15\" value=\"". $u . "\"><br/>
-    Password *<br /><input name=\"txtPass\" type=\"text\" size=\"30\" maxlength=\"32\" value=\"\"><br/>
+    Password *<br /><input name=\"txtPass\" type=\"password\" size=\"30\" maxlength=\"32\" value=\"\"><br/>
     <input name=\"btnSubmit\" type=\"submit\" value=\"Submit\"><br/>
     
     </form>
-- 
1.7.9.5

