From b53d6559293443e916f3415747cf904aa75ba751 Mon Sep 17 00:00:00 2001
From: root <root@softwaresecurity.(none)>
Date: Fri, 12 Apr 2013 00:03:33 +0200
Subject: [PATCH 16/23] Use individual mysql credentials

---
 includes/businessPage.inc.php |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/includes/businessPage.inc.php b/includes/businessPage.inc.php
index e38c3af..110b9af 100644
--- a/includes/businessPage.inc.php
+++ b/includes/businessPage.inc.php
@@ -13,8 +13,8 @@ session_start();
 $_BUSINESS = array();
 $_BUSINESS[ 'db_server' ] = 'localhost';
 $_BUSINESS[ 'db_database' ] = 'business';
-$_BUSINESS[ 'db_user' ] = 'root';
-$_BUSINESS[ 'db_password' ] = 'password';
+$_BUSINESS[ 'db_user' ] = 'bws';
+$_BUSINESS[ 'db_password' ] = 'Y_pwQZF2367w';
 
 // Declare the $html variable
 if(!isset($html)){
-- 
1.7.9.5

