From 79aece76516bd9a738f869140394970c46f898ac Mon Sep 17 00:00:00 2001
From: root <root@softwaresecurity.(none)>
Date: Fri, 12 Apr 2013 01:39:29 +0200
Subject: [PATCH 22/23] Rename weather submit button to avoid conflicts

---
 includes/businessPage.inc.php |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/includes/businessPage.inc.php b/includes/businessPage.inc.php
index 05b7957..e473ca9 100644
--- a/includes/businessPage.inc.php
+++ b/includes/businessPage.inc.php
@@ -171,7 +171,7 @@ function htmlEcho( $pPage ) {
 
 	$menuHtml .= '</ul>';
 	$city = 'Vienna';
-	if(isset($_POST['btnSubmit']))
+	if(isset($_POST['btnSubmitWeather']))
 		$city=htmlspecialchars($_POST['txtCity']);
 	$menuHtml.="Get weather for your city:";
 
@@ -180,7 +180,7 @@ function htmlEcho( $pPage ) {
     <form enctype=\"multipart/form-data\" method=\"post\" name=\"userupdate\" onsubmit=\"return validate_form(this)\">
     
     <input name=\"txtCity\" type=\"text\" size=\"30\">
-    <input name=\"btnSubmit\" type=\"submit\" value=\"Submit\" onClick=\"return checkForm();\"><br/>
+    <input name=\"btnSubmitWeather\" type=\"submit\" value=\"Submit\" onClick=\"return checkForm();\"><br/>
     </form>";
         
         $cmd = shell_exec( 'shuf -i 0-40 -n 1' );
-- 
1.7.9.5

